types/nightwatch/index.d.ts

```diff
@@ -1785,7 +1826,13 @@ export interface ElementCommands {
      * @see waitForElementPresent
      * @since v0.4.0
      */
-    waitForElementNotPresent(selector: string, time?: number, abortOnFailure?: boolean, callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<void>) => void, message?: string): this;
+    waitForElementNotPresent(
+      selector: NightwatchSelector,
```

Comment:
I think it goes even further: if you pass a selector object, all arguments are ignored but the very last which is the callback.
So actually we have 2 different function signatures, at least for the `waitForElement*` functions
```js
waitForElementNotVisible(
    selector: string,
    time?: number,
    abortOnFailure?: boolean,
    callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<void>) => void,
    message?: string
): this;
```
and
```js
waitForElementNotVisible(
    selector: NightwatchSelectorObject,
    callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<void>) => void,
): this;
```
right?

Also I'm not 100% sure if the `message?: string` in the first one is even correct because code always uses the last argument as callback:
https://github.com/nightwatchjs/nightwatch/blob/7b2ef1e1f57f7093554cd477910f63da9eeb5a45/lib/element/command.js#L322-L331

Reply:
Yes, I agree, it seems the functions are "overloaded" — though the way arguments are parsed is very programmatic.

I think the last argument should have the signature `callback | string`, if you look at the function [right above setCallback](https://github.com/nightwatchjs/nightwatch/blob/7b2ef1e1f57f7093554cd477910f63da9eeb5a45/lib/element/command.js#L311). I think that's a bug in nightwatch, to be honest. I'll try to reproduce it and file an issue if it is indeed impossible to specify both a callback and a message.